src/subgraph/actions/getMostRecentDomains.ts

```diff
@@ -0,0 +1,36 @@
+import { ApolloClient } from "@apollo/client/core";
+import { Domain } from "../../types";
+import * as queries from "../queries";
+import { DomainsQueryDto } from "../types";
+import { convertDomainDtoToDomain, performQuery } from "./helpers";
+
+export const getMostRecentDomains = async <T>(
+  apolloClient: ApolloClient<T>,
+  count: number = 100
+): Promise<Domain[]> => {
+  const domains: Domain[] = [];
+  let skip = 0;
+  let yetUnreceived = count;
+
+  while (true) {
+    const queryResult = await performQuery<DomainsQueryDto>(
+      apolloClient,
+      queries.getPastNDomains,
+      { count: yetUnreceived, startIndex: skip }
+    );
+
+    const queryResults = queryResult.data.domains;
+    for (const domain of queryResults) {
+      domains.push(convertDomainDtoToDomain(domain));
+    }
+
+    skip += queryResults.length;
```

Comment:
What happens when `queryResults.length == 0` but `yetUnreceived > 0`

Reply:
great point, added another break condition

Reply:
actually initially I was just gonna add a break condition but should I throw here? If you ask for 1000 and get 800 back and then the API craps out, should we consider that a valid exit case?

Reply:
I think valid exit is okay